simulator-spring-boot/src/main/java/org/citrusframework/simulator/web/rest/ScenarioResource.java

```diff
@@ -94,19 +97,20 @@ private synchronized void evictAndReloadScenarioCache(Set<String> scenarioNames,
      * @return the {@link ResponseEntity} with status {@code 200 (OK)} and the list of scenarios in body.
      */
     @GetMapping("/scenarios")
-    public ResponseEntity<List<Scenario>> getScenarios(@RequestParam(name = "nameContains", required = false) Optional< String> nameContains, @ParameterObject Pageable pageable) {
-        logger.debug("REST request get registered Scenarios, where name contains: {}", nameContains.orElse("*"));
+    public ResponseEntity<List<Scenario>> getScenarios(@RequestParam(name = "nameContains", required = false) Optional<String> nameContains, @ParameterObject Pageable pageable) {
+        var nameFilter = nameContains.map(contains -> decode(contains, UTF_8)).orElse("*");
```

Comment:
this could be split into a third commit, if wanted. fixes the filtering of scenarios where the name contains URL encoded values.